test/Autofac.Specification.Test/Lifetime/AsyncDisposeProvidedInstanceTests.cs

```diff
@@ -0,0 +1,209 @@
+            scope.Resolve<AsyncOnlyDisposeTracker>();
+
+            await scope.DisposeAsync();
+            Assert.Equal(1, count);
+
+            await container.DisposeAsync();
+            Assert.Equal(1, count);
+        }
+
+        [Fact]
+        public async Task ResolvedProvidedInstances_NotOwnedByLifetimeScope_NeverDisposed()
+        {
+            var builder = new ContainerBuilder();
+            var disposable = new AsyncOnlyDisposeTracker();
+            builder.RegisterInstance(disposable).ExternallyOwned();
+            var container = builder.Build();
+            container.Resolve<AsyncOnlyDisposeTracker>();
+
+            await container.DisposeAsync();
+
+            Assert.False(disposable.IsAsyncDisposed);
+        }
+
+        [Fact]
+        public async Task ResolvedProvidedInstances_NotOwnedByNestedLifetimeScope_NeverDisposed()
+        {
+            var builder = new ContainerBuilder();
+            var disposable = new AsyncOnlyDisposeTracker();
+            var container = builder.Build();
+            var scope = container.BeginLifetimeScope(b => b.RegisterInstance(disposable).ExternallyOwned());
+            scope.Resolve<AsyncOnlyDisposeTracker>();
+
+            await scope.DisposeAsync();
+            Assert.False(disposable.IsAsyncDisposed);
+
+            await container.DisposeAsync();
+            Assert.False(disposable.IsAsyncDisposed);
+        }
+
+        [Fact]
+        public async Task UnresolvedProvidedInstances_DisposedWithLifetimeScope()
+        {
+            var builder = new ContainerBuilder();
+            var disposable = new AsyncOnlyDisposeTracker();
+            builder.RegisterInstance(disposable);
+            var container = builder.Build();
+
+            await container.DisposeAsync();
+
+            Assert.True(disposable.IsAsyncDisposed);
+        }
+
+        [Fact]
+        public async Task UnresolvedProvidedInstances_DisposedWithLifetimeScope_OnlyDisposedOnce()
+        {
+            var builder = new ContainerBuilder();
+            var count = 0;
+            var disposable = new AsyncOnlyDisposeTracker();
+            disposable.Disposing += (sender, e) => count++;
+            builder.RegisterInstance(disposable);
+            var container = builder.Build();
+
+            await container.DisposeAsync();
+
+            Assert.Equal(1, count);
+        }
+
+        [Fact]
+        public async Task UnresolvedProvidedInstances_DisposedWithNestedLifetimeScope()
+        {
+            var builder = new ContainerBuilder();
+            var disposable = new AsyncOnlyDisposeTracker();
+            var container = builder.Build();
+            var scope = container.BeginLifetimeScope(b => b.RegisterInstance(disposable));
+
+            await scope.DisposeAsync();
+
+            Assert.True(disposable.IsAsyncDisposed);
+        }
+
+        [Fact]
+        public async Task UnresolvedProvidedInstances_ActualAsyncDisposable_CorrectlyDisposed()
```

Comment:
The name on this test reads weird. It sounds like I should be looking for the instance to be disposed, but in this case it's `ExternallyOwned` and not disposed. Looks like it's a copy/paste of `ResolvedProvidedInstances_NotOwnedByNestedLifetimeScope_NeverDisposed` up on line 93?

Reply:
Yeah, that's just me getting the test wrong.